### ISSUE TITLE
Usability improvements to Table View

### ISSUE THREAD

**Issue by Ig-Rebollo** [kobotoolbox/kpi#3199]
This issue if the second (and optional) half of a series improvements to the table view UI. These improvements are divided in two groups:

1. **Styling** improvements, (issue #3198) which should be done before any other table view related additions; and
2. **Usability** improvements, (this issue) which can be done along with those updates, or later on, depending on timelines and availability.

### Usability improvements
The following updates involve small new functionalities and minor additions to make the experience more user friendly. These are not as urgent as the accessibility improvements, and most are optional, depending on what we have time for. Many can also be postponed, in which case they will be recorded in the Notion idea repository. See list below.
![TableviewNEW](https://user-images.githubusercontent.com/22472326/118162868-428ca000-b3ef-11eb-8d4e-b9c8ab0df917.jpg)

**Making column headers clearer and more useful**
- [ ] Include an arrow with a dropdown menu for easier to understand actions (current sorting is not very intuitive). Note that a sorted column would stay highlighted and include the icon of the type of sorting. The dropdown would also allow you to cancel the sorting. See animation:
![column sorting](https://user-images.githubusercontent.com/22472326/117689363-aa40b200-b187-11eb-8b8b-c8076312a1bc.gif)

**Include more navigation options on top.**
- [ ] While 'filter' might be something more challenging for a later occasion, we could start with a 'hide columns' functionality. This could also be accessible from the column header dropdown described before.
- [ ] Note that in order to do this, 'number of results' and 'rows displayed' should be removed from where they are and included in the table itself.
![hide fields](https://user-images.githubusercontent.com/22472326/117689752-14f1ed80-b188-11eb-9da5-c406298c9cc0.gif)

Additional details and specs can be found [in the following Figma file](https://www.figma.com/file/vbHVtRkdit3dqd5y8JFfXF/Table-view-redesign?node-id=0%3A1).
